Description: Calling pm-powersave before pm-suspend is very slow, it. e. g. takes > 1.5 seconds on a Nexus 7. This does not nearly compensate the potentially slightly slower suspend operation in low power mode, so disable it.
Author: Martin Pitt <martin.pitt@ubuntu.com>

Index: pm-utils/pm/sleep.d/00powersave
===================================================================
--- pm-utils.orig/pm/sleep.d/00powersave	2014-05-07 11:40:22.623850566 +0200
+++ pm-utils/pm/sleep.d/00powersave	2014-05-07 11:40:41.659849848 +0200
@@ -2,12 +2,17 @@
 
 . "${PM_FUNCTIONS}"
 
+# do not run pm-powersave on ARM during suspend; the 1.5 seconds that it takes
+# to run it don't nearly compensate the potentially slightly slower suspend
+# operation in low power mode
+ARCH=`uname -m`
+
 command_exists pm-powersave || exit $NA
 
 case $1 in
-    suspend|hibernate) pm-powersave false ;;
+    suspend|hibernate) [ "$ARCH" != "${ARCH#arm}" ] || pm-powersave false ;;
     resume|thaw)       pm-powersave ;;
     *) exit $NA ;;
 esac
 exit 0
-    
\ No newline at end of file
+    
